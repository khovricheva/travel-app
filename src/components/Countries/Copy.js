import React, { useState } from 'react';
import Button from '@material-ui/core/Button';

const Copy = ({ copy, code, link }) => {
  const [state, setState] = useState(false);

  const handleClick = () => {
    setState(true);
    navigator.clipboard.writeText(window.location.href + link.slice(1));
    let timer = setTimeout(() => {
      setState(false);
      clearTimeout(timer);
    }, 1000);
  };

  return (
    <Button
      size="small"
      color="inherit"
      variant="outlined"
      onClick={handleClick}
    >
      {!state ? copy[code] : copy.copied[code]}
    </Button>
  );
};

export default Copy;
