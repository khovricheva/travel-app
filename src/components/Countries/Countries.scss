.countries {
  display: grid;
  grid-template-columns: 1fr;
  padding-right: 5px;
  padding-left: 5px;

  .cardContainer {
    display: flex;
    justify-content: center;
    margin-bottom: 10px;

    .card {
      width: 90%;
      height: auto;
      max-width: 35rem;
      min-height: 12rem;
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      color: white;
      cursor: pointer;
      transition: transform 0.3s;

      .countryImage::after {
        content: '';
        position: absolute;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        background: linear-gradient(
          to top,
          rgba(0, 0, 0, 0.548),
          rgba(0, 0, 0, 0.25)
        );
      }

      .link {
        text-decoration: none;
        color: white;
      }
    }

    .card:hover {
      transform: scale(1.1);
    }
  }
}

@media screen and (min-width: 500px) {
  .countries {
    display: grid;
    grid-template-columns: 1fr;
    .cardContainer {
      display: flex;
      justify-content: center;
      margin-bottom: 10px;
      .card {
        width: 90%;
        height: auto;
        max-width: 40rem;
        min-height: 18rem;
      }
    }
  }
}

@media screen and (min-width: 768px) {
  .countries {
    display: grid;
    grid-template-columns: 1fr 1fr;
    .cardContainer {
      display: flex;
      justify-content: center;
      margin-bottom: 10px;
      .card {
        width: 90%;
        height: auto;
        max-width: 30rem;
        min-height: 15rem;
      }
    }
  }
}

@media screen and (min-width: 1100px) {
  .countries {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media screen and (min-width: 1800px) {
  .countries {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
  }
}