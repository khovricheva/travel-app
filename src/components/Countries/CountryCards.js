import React from 'react';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import { useSelector } from 'react-redux';
import NotFound from '../NotFound';
import { Link } from 'react-router-dom';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import { Button } from '@material-ui/core';
import translate from '../../translate';
import Spinner from '../Spinner/Spinner';

const CountryCards = (props) => {
  const { countries } = props;
  const code = useSelector((state) => state.code);

  if (countries === 'NotFound') {
    return <NotFound />;
  }

  if (countries.length === 0) {
    return <Spinner />;
  }

  return (
    <div className="countries">
      {countries.map((item) => (
        <Box key={item.name.en} my={4} className="cardContainer">
          <Card className="card">
            <CardMedia
              media="picture"
              className="countryImage"
              alt="Country Photo"
              image={item.introPhoto}
              title="Country Photo"
            />
            <CardContent
              style={{ position: 'relative', backgroundColor: 'transparent' }}
            >
              <Typography gutterBottom variant="h5" component="h2">
                {item.name[code]}
              </Typography>
              <Typography variant="body2" component="p">
                {item.capital[code]}
              </Typography>
            </CardContent>
            <CardActions style={{ position: 'relative' }}>
              <Link
                className='link'
                to={`/country/${item.name.en.toLowerCase()}`}
              >
                <Button size="small" color="inherit" variant="outlined">
                  {translate.learnMore[code]}
                </Button>
              </Link>
            </CardActions>
          </Card>
        </Box>
      ))}
    </div>
  );
};

export default CountryCards;
