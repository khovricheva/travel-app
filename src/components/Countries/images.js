const imageArray = [
  { img: '/images/Canada.jpeg', countryName: 'Canada', capital: 'Ottawa' },
  { img: '/images/Australia.jpeg', countryName: 'Australia', capital: 'Canberra' },
  { img: '/images/France.jpeg', countryName: 'France', capital: 'Paris' },
  { img: '/images/Greece.jpeg', countryName: 'Greece', capital: 'Athens' },
  { img: '/images/Netherlands.jpeg', countryName: 'Netherlands', capital: 'Amsterdam' },
  { img: '/images/Spain.jpeg', countryName: 'Spain', capital: 'Madrid' },
  { img: '/images/Ukraine.jpeg', countryName: 'Ukraine', capital: 'Kyiv' },
  { img: '/images/USA.jpeg', countryName: 'USA', capital: 'Washington' },
];

export default imageArray;
