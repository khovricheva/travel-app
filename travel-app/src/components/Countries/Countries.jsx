import React from 'react';
import './Countries.scss';

const Countries = () => {
  return <div></div>;
};

export default Countries;
