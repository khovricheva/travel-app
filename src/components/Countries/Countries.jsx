import React, { useContext } from 'react';
import './Countries.scss';
import { CountryContext } from '../../context/GlobalContext';
import CountryCards from './CountryCards';
import ErrorFallback from '../ErrorBoundary';
import { ErrorBoundary } from 'react-error-boundary';

const Countries = () => {
  const countries = useContext(CountryContext);

  return (
    <ErrorBoundary
      FallbackComponent={ErrorFallback}
      onReset={() => {
        window.location.reload();
        localStorage.clear();
      }}
    >
      <CountryCards countries={countries} />
    </ErrorBoundary>
  );
};

export default Countries;
