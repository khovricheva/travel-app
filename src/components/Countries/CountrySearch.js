import React, { useContext } from 'react';
import { CountryContext } from '../../context/GlobalContext';
import { useLocation } from 'react-router-dom';
import CountryCards from './CountryCards';
import ErrorFallback from '../ErrorBoundary/ErrorBoundary';
import { ErrorBoundary } from 'react-error-boundary';

const countryFilter = (countries, name) => {
  const result = countries.filter(
    (item) =>
      item.name.en.toLowerCase().includes(name) ||
      item.capital.en.toLowerCase().includes(name) ||
      item.name.ru.toLowerCase().includes(name) ||
      item.capital.ru.toLowerCase().includes(name) ||
      item.name.ua.toLowerCase().includes(name) ||
      item.capital.ua.toLowerCase().includes(name)
  );
  return result.length === 0 ? 'NotFound' : result;
};

const CountrySearch = () => {
  const countries = useContext(CountryContext);
  const location = useLocation();

  const name = location.pathname.slice(13).toLowerCase();
  const countryResult = countryFilter(countries, name);
  return (
    <ErrorBoundary
      FallbackComponent={ErrorFallback}
      onReset={() => {
        window.location.reload();
        localStorage.clear();
      }}
    >
      <CountryCards countries={countryResult} />{' '}
    </ErrorBoundary>
  );
};

export default CountrySearch;
