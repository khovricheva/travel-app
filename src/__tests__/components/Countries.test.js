import React from 'react';
import Countries from '../../components/Countries/Countries';
import { render, screen } from '../test-utils';

describe('App component', () => {
  const countries = [
    {
      name: {
        en: 'Austria',
        ru: 'Австрия',
        ua: 'Австрія',
      },
      capital: {
        en: 'Vienna',
        ru: 'Вена',
        ua: 'Відень',
      },
      flag: 'https://travel-api-git-main-imbatman.vercel.app/flag/1.svg',
      introPhoto:
        'https://travel-api-git-main-imbatman.vercel.app/countriesPhoto/1.jpg',
    },
  ];
  it('Renders correctly', () => {
    const wrapper = render(<Countries countries={countries} />);
    expect(wrapper).toMatchSnapshot();
  });
});
