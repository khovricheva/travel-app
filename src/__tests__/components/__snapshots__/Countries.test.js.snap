// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="countries"
      >
        <div
          class="MuiBox-root MuiBox-root-1 cardContainer"
        >
          <div
            class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              alt="Austria"
              class="MuiCardMedia-root countryImage"
              media="picture"
              style="background-image: url(https://travel-api-git-main-imbatman.vercel.app/countriesPhoto/1.jpg);"
              title="Austria"
            />
            <div
              class="MuiCardContent-root"
              style="position: relative; background-color: transparent;"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Austria
              </h2>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Vienna
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing"
              style="position: relative;"
            >
              <a
                class="link"
                href="/country/austria"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Learn More
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="countries"
    >
      <div
        class="MuiBox-root MuiBox-root-1 cardContainer"
      >
        <div
          class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            alt="Austria"
            class="MuiCardMedia-root countryImage"
            media="picture"
            style="background-image: url(https://travel-api-git-main-imbatman.vercel.app/countriesPhoto/1.jpg);"
            title="Austria"
          />
          <div
            class="MuiCardContent-root"
            style="position: relative; background-color: transparent;"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Austria
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Vienna
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
            style="position: relative;"
          >
            <a
              class="link"
              href="/country/austria"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Learn More
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
